sstable/colblk_writer.go

```diff
@@ -443,6 +444,10 @@ func (w *RawColumnWriter) AddWithBlobHandle(
 		return err
 	}
 	w.props.NumValuesInBlobFiles++
+	if err := w.blobRefLivenessIndexBlock.maybeAddNewState(h.ReferenceID, h.BlockID); err != nil {
+		return err
+	}
+	w.blobRefLivenessIndexBlock.addLiveValue(h.ReferenceID, h.BlockID, h.ValueID, uint64(h.ValueLen))
```

Comment:
thoughts on moving the call to `maybeAddNewState` into `addLiveValue`? Then we can hide a bit more of the internals of the liveness index block. Unfortunately since this is a hot path, I also think we should lift the check to see whether we need to add a new state also into addLiveValue. That would ensure that in the common happy case that we've already seen the referenceID, we don't need to pay the cost of a function call

Reply:
> That would ensure that in the common happy case that we've already seen the referenceID, we don't need to pay the cost of a function call

ah cool that makes sense; done